hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/pipeline/transform/AbstractTransform.java

```diff
@@ -115,4 +131,31 @@ protected final boolean shouldRebalanceAnyInput() {
         }
         return false;
     }
+
+    /**
+     * Determines the local parallelism value for the transform by looking
+     * its local parallelism, preferred local parallelism, and the default
+     * local parallelism provided in Pipeline.Context object.
+     * <p>
+     * If none of them is set, returns the default local parallelism
+     * provided in Pipeline.Context object.
+     */
+    protected void determineLocalParallelism(int localParallelism, int preferredLocalParallelism, Pipeline.Context ctx) {
```

Comment:
The `localParallelism` parameter is always the `localParallelism()` method, we can remove this parameter.